Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -1613,19 +1613,21 @@ public void cleverTapSetLocation(){
     }
 
     @JavascriptInterface
-    public void showDialer(String phoneNum) {
+    public void showDialer(String phoneNum, boolean call) {
         Intent intent = new Intent();
-        intent.setAction(Intent.ACTION_CALL);
-        phoneNumber = phoneNum;
-        if (ContextCompat.checkSelfPermission(activity, Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
-            ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.CALL_PHONE}, REQUEST_CALL);
-        } else {
-            if (intent != null) {
-                phoneNumber = "tel:" + phoneNum;
+        phoneNumber = "tel:" + phoneNum;
+        if (call) {
+            intent.setAction(Intent.ACTION_CALL);
+            if (ContextCompat.checkSelfPermission(activity, Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(activity, new String[]{Manifest.permission.CALL_PHONE}, REQUEST_CALL);
+            } else {
                 intent.setData(Uri.parse(phoneNumber));
                 activity.startActivity(intent);
             }
-
+        } else {
+            intent.setAction(Intent.ACTION_DIAL);
+            intent.setData(Uri.parse(phoneNumber));
```

Comment:
these if else conditions can be reduced further
`            intent.setData(Uri.parse(phoneNumber));
            activity.startActivity(intent);`
            These 2 line can be moved outside
            And for `intent.setAction(Intent.ACTION_CALL);` you can use ternary operator